docs/porting/connectivity/ble.md

```diff
@@ -0,0 +1,37 @@
+# Bluetooth Low Energy (BLE)
+
+Mbed OS uses the Cordio stack for its BLE host. We recommend porting the Cordio platform<!--is that what they call it?--> to your board, or porting at the HCI level<!--what's the distinction between porting the platform and porting the HCI level? If one is ported to the board, where is the other ported to? The chip? What's the difference between porting to a whole board v porting to a chip?--> as described in the HCI documentation.
```

Comment:
API level - you port the whole stack, and provide the same interface mbed ble provides.
HCI level - the stack is split into two parts Host and Controller, HCI stands for host controller interface, the communication across the HCI is well defined and as long as you handle all the messages you don't care who the host is and viceversa, if you have a chip with its own controller then that's the level you want to port at (like wb55)
Cordio level - you use cordio for everything and you port the platform layer of cordio so that cordio can run on your board directly like the nrf52

Reply:
So the API level port is what we describe doing on this page, and the Cordio and HCI level ports are what the [Cordio docs](https://os.mbed.com/docs/mbed-cordio/19.02/introduction/index.html) describe?

Reply:
Yes. I don't expect API porting to be in high demand, it's just here as an option since there's no reason to stop people from doing it but I expect ports to use Cordio.